tests/test_kernels.py

```diff
@@ -0,0 +1,89 @@
+"""Kernel tests."""
+
+from typing import Any
+
+import numpy as np
+import torch
+from attrs import asdict, has
+from hypothesis import given
+
+from baybe.kernels.base import Kernel
+from baybe.kernels.composite import AdditiveKernel, ProductKernel, ScaleKernel
+
+from .hypothesis_strategies.kernels import kernels
+
+# TODO: Consider deprecating these attribute names to avoid inconsistencies
+_RENAME_DICT: dict[str, str] = {
+    "base_kernels": "kernels",
+}
+"""Dictionary for resolving name differences between BayBE and GPyTorch attributes."""
+
+
+def validate_kernel_components(obj: Any, mapped: Any, **kwargs) -> None:
+    """Validate that all kernel components are translated correctly.
+
+    Args:
+        obj: An object occurring as part of a BayBE kernel.
+        mapped: The corresponding object in the translated GPyTorch kernel.
+        **kwargs: Optional kernel arguments that were passed to the GPyTorch kernel.
+    """
+    # Assert that the kernel kwargs are correctly mapped
+    # TODO: Consider introducing intermediate kernel classes or class variables
```

Comment:
Thoughts here?

Reply:
Intermediate classes feels a bit like an overkill here, in particular if the only current purpose where we need them is making this check here easier.

Reply:
Don't see the necessity that they need to share something, tbh. We have base classes with no content all over the place, so why the requirement here? But we can also leave it as is, I don't mind

Reply:
To be honest, I always found base classes with basically no intermediate content a bit weird 😆 But you are right, it might be more in-line with our design in general.
Would having such intermediate classes be useful in other places as well? In that case I'd say definitely go for it, other wise I am also quite open/leave it up to you.

Reply:
intermediate makes sense even when empty if the class info itself is the property we want to distinguish
here it would also be possible to include a mechanism (similar as done for `target` and `targets`)so the intermediate class gets the code for `base_kernels`

Reply:
done